modules/bridge/src/main/ts/ephox/bridge/components/toolbar/ToolbarSplitButton.ts

```diff
@@ -39,6 +40,7 @@ export interface ToolbarSplitButton {
   onSetup: (api: ToolbarSplitButtonInstanceApi) => (api: ToolbarSplitButtonInstanceApi) => void;
   onAction: (api: ToolbarSplitButtonInstanceApi) => void;
   onItemAction: (api: ToolbarSplitButtonInstanceApi, value: string) => void;
+  readonly: boolean;
```

Comment:
```suggestion
  readonlyEnabled: boolean;
```
`readonly` implies the component will be readonly, we should use something more accurate. Maybe `enabledInReadonly`?

Reply:
Good call, I'll make the required changes.

Reply:
That relates to my comment above; I don't know if `readonlyEnabled` is right either. Maybe we need a quick API/architecture discussion.

Reply:
I've replaced `readonly: boolean` property with `allowedModes: string[]` so the registered allowedModes is used to match the editor mode to determine if the button state.